rajawali/src/main/java/c/org/rajawali3d/core/ARenderControl.java

```diff
@@ -0,0 +1,422 @@
+                    sceneViews.add(sceneView);
+                    // TODO depth order implementation (sort the list?)
+                    sceneView.setViewportDepthOrder(viewportDepthOrder);
+                    sceneView.onAddToRenderControl(ARenderControl.this);
+                }
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    @Override
+    public boolean removeSceneView(@NonNull final SceneView sceneView) {
+        final RenderTask task = new RenderTask() {
+            @Override
+            protected void doTask() {
+                if (sceneViews.contains(sceneView)) {
+                    sceneViews.remove(sceneView);
+                    sceneView.onRemoveFromRenderControl();
+                }
+            }
+        };
+        return executeRenderTask(task);
+    }
+
+    //
+    // SurfaceRenderer methods
+    //
+
+    @RenderThread
+    @Override
+    @CallSuper
+    public void onRenderContextAcquired(RenderContextType renderContextType,
+                                        int renderContextMajorVersion, int renderContextMinorVersion) {
+        RenderContext.init(renderContextType, renderContextMajorVersion, renderContextMinorVersion);
+
+        setRenderThread();
+
+        setFrameRate(frameRate);
+    }
+
+    @RenderThread
+    @Override
+    @CallSuper
+    public void onSurfaceSizeChanged(int width, int height) {
+        surfaceSize = new SurfaceSize(width, height);
+
+        if (isInitialSurfaceSize) {
+            isInitialSurfaceSize = false;
+            startFrames();
+            renderControlClient.onRenderControlAvailable(this, surfaceSize);
+        } else {
+            renderControlClient.onSurfaceSizeChanged(surfaceSize);
+        }
+    }
+
+    @RenderThread
+    @Override
+    public void onRenderFrame() {
+
+        if (!areFramesEnabled()) {
+            return;
+        }
+
+        final double deltaTime = getFrameDeltaTime();
+
+        try {
+            onFrameStart(deltaTime);
+
+            renderViews();
+
+            onFrameEnd(deltaTime);
+
+        } catch (InterruptedException e) {
+            Log.e(TAG, "onRenderFrame(): Frame incomplete due to thread interruption.");
+        }
+
+        updateFPS();
+    }
+
+    @RenderThread
+    private double getFrameDeltaTime() {
+        long currentTime = System.nanoTime();
+        double deltaTime = (currentTime - lastFrameTime) / 1e9;
+        lastFrameTime = currentTime;
+
+        return deltaTime;
+    }
+
+    //TODO Override this method in a stereo renderer to render views for each Eye
+    @RenderThread
+    private void renderViews() throws InterruptedException {
+        // Propagate to SceneViews...//
+        for (int i = 0, j = sceneViews.size(); i < j; i++) {
+            ((ASceneView)sceneViews.get(i)).onRender();
+        }
+    }
+
+    @RenderThread
+    private void updateFPS() {
+        ++frameCount;
+        if (frameCount % 50 == 0) {
```

Comment:
Well, specifically if the goal is to update once per second, it should be the operating frame rate.